example/CMakeLists.txt

```diff
@@ -156,3 +160,10 @@ set(DOXYGEN_CONF_IN "doxygen.conf")
 include("${PROJECT_SOURCE_DIR}/../Modules/defaultDocuTarget.cmake")
 
 
+IF( EXISTS "${CMAKE_CURRENT_BINARY_DIR}/compile_commands.json" )
+  EXECUTE_PROCESS( COMMAND ${CMAKE_COMMAND} -E copy_if_different
+    ${CMAKE_CURRENT_BINARY_DIR}/compile_commands.json
+    ${CMAKE_CURRENT_SOURCE_DIR}/compile_commands.json
```

Comment:
Why do you want to copy that into the source directory? Normally generated files should stay in the build folder in order to not pollute the source directory. Instead, you could configure your plugin to read the `compile_commands.json` from the build folder.